### PR TITLE
Add `oats-fetch-adapter`

### PR DESCRIPTION
**Pull request by ganymedes01**
moved over from https://github.com/smartlyio/oats-fetch-adapter which was kinda abandoned so there was some heavy refactoring done as well

### CODE REVIEW

File: test/fetch-adapter/fetch-adapter.spec.ts
Comment (valentin-nemcev) [inline test/fetch-adapter/fetch-adapter.spec.ts:1]: I see you test that it sends stuff to server correctly, but I don't see tests that it receives stuff correctly, beyond 200 status?

Comment (ganymedes01) [reply test/fetch-adapter/fetch-adapter.spec.ts:1]: true. i’ll add those

File: packages/oats-fetch-adapter/index.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    const server = arg.servers[0];
+    const data = toRequestData(arg.body);
+    const url = new URL(server + arg.path);
+    const requestContentType = arg.body?.contentType;
+
+    Object.entries(arg.query ?? {})
+      .filter(([, value]) => !!value)
+      .forEach(([key, value]) => url.searchParams.append(key, `${value}`));
+
+    const response = await fetch(
```
Comment (valentin-nemcev) [inline packages/oats-fetch-adapter/index.ts:74]: do you need to put credentials: "some-origin" or whatever you need to send cookies with fetch? And some other security stuff about cors? I always forget

Comment (valentin-nemcev) [reply packages/oats-fetch-adapter/index.ts:74]: ok, nvm, it has defaults

**Review by sakari — APPROVED** [review]
I guess this is fine if it works

**Pull request merged**